object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/KnnPredicate.java

```diff
@@ -40,12 +40,23 @@ public ValueExpr getChild() {
 
    @Override
    public String toString() {
-      return leftChild.toString() + "<-> " + vector + "~" + knn;
+      return toQueryString();
    }
 
    @Override
    public String toQueryString() {
-      return leftChild.toQueryString() + "<-> " + vector + "~" + knn;
+      StringBuilder builder = new StringBuilder(leftChild.toQueryString());
```

Comment:
To clarify the idea is to have a toQueryString method that takes a StringBuilder as an argument. So then all the nested child predicates can write to the StringBuilder instead of creating a string for each level. It is a bigger undertaking, maybe we should move it to a separate JIRA?

Reply:
Makes sense. Let's do that in this contribution, since it is quite small :P

Reply:
@wburns the change for introducing at any level the string builder has a larger scope, so I opened https://issues.redhat.com/browse/ISPN-15932 for it.